### PR TITLE
Fix numeric widget

### LINKED ISSUE #1836: Numeric Param Widgets, initialised with None, displayed as active "0"
#### ALL software version info
Panel 10.1

#### Description of expected behavior and the observed behavior
Widgets or parameters initialised as "None" has value of 'None', but still display an active '0' 

#### Complete, minimal, self-contained example code that reproduces the issue

```
class Test(param.Parameterized) :
    w_none_but_0 = param.Number(default=None, allow_None=True)
    w_none_forced = param.Number(default=None, allow_None=True)

    
    # Since default None, was not displayed correctly, tried to override with None,
    # to clear the widget, but had no effect..
    def __init__(self, **params):
        super().__init__(**params)
        self.param.set_param("w_none_forced", 450) 
        self.param.set_param("w_none_forced", None)    # didn't improve anything
          
    
t = Test()
pn.Row(t)

```

#### Screenshots or screencasts of the bug in action
image

### PR DESCRIPTION
**Pull request by Hoxbro**
Fixes #1836

 I'm a bit uncertain if the change in `panel/param.py` is okay, since I don't know if every widgets accepts None as a value. If there is a better way to implement this, this PR can be closed.

Example of fix:
``` python
import param
import panel as pn


class Test(param.Parameterized) :
    number = param.Number(default=0, allow_None=True)
    none = param.Number(default=None, allow_None=True)

    def view(self):
        return self.number, self.none


t = Test()
pn.Column(t, t.view).servable()
```

### CODE REVIEW

File: panel/widgets/tables.py [hunk 1]
```diff
@@ -633,7 +633,7 @@ class DataFrame(BaseTable):
     hierarchical = param.Boolean(default=False, constant=True, doc="""
         Whether to generate a hierachical index.""")
 
-    fit_columns = param.Boolean(default=None, doc="""
```
Comment (philippjfr) [inline panel/widgets/tables.py:636]: This seems unrelated.

**Comment by philippjfr** [comment]
I didn't think you could pass None to a slider. What happens?

File: panel/widgets/tables.py (hunk 1 above)
Comment (Hoxbro) [reply panel/widgets/tables.py:636]: The changes i made gave two type of errors, though both related to `controls(jslink=True)`.  One of the error was related to `pn.indicators.Progress` and the other `pn.widgets.DataFrame`. The one for DataFrame can be reproduced by:
```python
df_widget = pn.widgets.DataFrame()
df_widget.controls(jslink=True)
# ValueError: Boolean 'value' only takes a Boolean value.
```

I could see in the [documentation](https://panel.holoviz.org/reference/widgets/DataFrame.html) that it should be True by default.

**Comment by Hoxbro** [comment]
It works but it is in no way perfect.
example

The reason for the changes to the sliders is that `pn.indicators.Progress` `value` need to be able to give a non integer value. I get an error by running the following without setting `allow_None=True`. 

```
indeterminate = pn.indicators.Progress()
indeterminate.controls(jslink=True)
# ValueError: Parameter 'value' must be an integer.
```

If there is a better way to get it working, I would definitely like to here it.

**Comment by philippjfr** [comment]
Revived this in #3174.
